### DISCUSSION TITLE
Lazy loading components

### DISCUSSION THREAD

**Discussion by ciarantr** [nuxt/nuxt#17853] [Questions] [answered]
Hi, I'm having issues with lazy loading components with nuxt 3. An 500 error is given with the message: Invalid value used with weak key map.

For instants if I ass the prefix Lazy to the footer

Any advice on where I'm going wrong would be greatly appreciated.

**Comment by danielroe** [comment]
If you are using `<LazyFooter />` and it isn't working, would you open an issue with a reproduction? If you're using dynamic components, check out https://github.com/nuxt/framework/discussions/2883.

**Comment by ciarantr** [comment, marked as answer]
Thanks for the quick response. I just want to clarify incase I have something configured incorrectly before I commit an issue. 

I have a default layout page structured as
`<heder>
<slot>
<LazyTheFooter/>
`

To my understaning I simply prefixed the component TheFooter.vue with "Lazy" & the filename remains TheFooter.vue?

**Reply by danielroe** [reply]
Yes, that _should_ work 👍

**Reply by ciarantr** [reply]
Ok great, it does not. 

[Vue warn]: Failed to resolve component: LazyTheFooter
If this is a native custom element, make sure to exclude it from component resolution via compilerOptions.isCustomElement. 
  at <Default > 
  at <AsyncComponentWrapper > 
  at <BaseTransition mode="out-in" appear=false persisted=false  ... > 
  at <Transition name="layout" mode="out-in" > 
  at <NuxtLayout> 
  at <App> 
  at <NuxtRoot> runtime-core.esm-bundler.js:38


Thanks for taking the time to respond. I will open an issue later with a reproduction.

**Reply by mihail727** [reply]
Have you tried restarting `npm run dev`? 
It often happens to me that if I create a new component while running `npm run dev` and try to visit lazy on it - such an error appears - just restarting helps

**Reply by ciarantr** [reply]
Hi @mihail727 , yes I have tried to restart the server multiple times. No luck unfortunately.

**Reply by ciarantr** [reply]
Ok, Just learned that upgrading nuxt3 to latest has caused they issue rolling back to previous version.

Thanks for all the help @danielroe & @mihail727

**Reply by danielroe** [reply]
If you believe it to be a bug, please raise an issue.

**Reply by 7ute** [reply]
Had the same problem when updating from [3.0.0-27418641.56aabd6](https://www.npmjs.com/package/nuxt3/v/3.0.0-27418641.56aabd6) to [3.0.0-27419629.87eb7d0](https://www.npmjs.com/package/nuxt3/v/3.0.0-27419629.87eb7d0)
and appears to be related to https://github.com/nuxt/framework/pull/3305
Adding the `global` bool to the component config seems to fix it.
See: https://github.com/nuxt/framework/discussions/2883#1142127837

**Reply by little3201** [reply]
same question

**Comment by sjonibek** [comment]
I have "TheFooter" component too. When I change to "LazyTheFooter" an 500 error is given with the message: Invalid value used with weak key map.

**Reply by danielroe** [reply]
cc: @antfu

**Comment by ciarantr** [comment]
Hi all, I believe @7ute answered this issue. see issue #3305. When using dynamic imports you will need to update your nuxt config and add the following.

`
	components: {
		global: true,
		dirs: ['~/components']
	},
`

I'm not sure if this is a fix & the documentation needs to be updated when adding Lazy prefix to components.
There was an issue raised but I cannot seen to find it.

**Reply by danielroe** [reply]
We shouldn't need this when using lazy components - that is likely a regression.

**Reply by ciarantr** [reply]
Ok, I will add this to the issues and leave a link to the thread.
